### PR TITLE
community[patch]: Add grammar support for node-llama-cpp LLM

### PR DESCRIPTION
**Pull request by Shaydesh**
Hello, I am attempting to add a configuration to langchainjs node-llama-cpp implementation that will make it respond in a specific format like JSON or a list.  This is a functionality that the python langchain has for llama.  The functionality for this is documented here: https://withcatai.github.io/node-llama-cpp/guide/grammar .

### CODE REVIEW

**Review by bracesproul — COMMENTED** [review]
This looks really good! I left one comment, and I believe you'll need to run `yarn format && yarn lint:fix` to resolve any prettier/eslint issues.

Also, if you could add an integration test for this that would be much appreciated!

Please re-request my review once those changes have been pushed so I can get this merged

File: libs/langchain-community/src/llms/llama_cpp.ts [hunk 1]
```diff
@@ -65,9 +75,13 @@ export class LlamaCpp extends LLM<LlamaCppCallOptions> {
     this.topK = inputs?.topK;
     this.topP = inputs?.topP;
     this.trimWhitespaceSuffix = inputs?.trimWhitespaceSuffix;
+    this.jsonSchema = inputs?.jsonSchema;
+    this.gbnf = inputs?.gbnf;
     this._model = createLlamaModel(inputs);
     this._context = createLlamaContext(this._model, inputs);
     this._session = createLlamaSession(this._context);
+    this._jsonSchema = createLlamaJsonSchemaGrammar(this.jsonSchema);
+    this._gbnf = createCustomGrammar(this.gbnf);
```
[the remark this answers is not in the document]
Comment (Shaydesh) [reply libs/langchain-community/src/llms/llama_cpp.ts:91]: Yes, I will do this! Thank you for responding.  Sorry I didn't see your comments before my last update.

File: libs/langchain-community/src/llms/llama_cpp.ts [hunk 2]
```diff
@@ -48,12 +57,20 @@ export class LlamaCpp extends LLM<LlamaCppCallOptions> {
 
   trimWhitespaceSuffix?: boolean;
 
+  jsonSchema?:  object;
+
+  gbnf?: string;
+
```
Comment (bracesproul) [inline libs/langchain-community/src/llms/llama_cpp.ts:63]: Unused, drop
```suggestion
```

**Review by bracesproul — APPROVED** [review]
lgtm! One more question, could this be applied to the chat model integration too? I'll merge this now, but it would be great to get it in for chat models too since those are more widely used.

**Pull request merged**